### PR TITLE
NS2.1: System.IO, System.Reflection.Emit, etc

### PR DESCRIPTION
**Pull request by EgorBo**
~~Needs: https://github.com/mono/corefx/pull/251~~
Contributes to: https://github.com/mono/mono/issues/11430

~~Current status: https://gist.github.com/EgorBo/69c3c4f2192495b302c7bb911d80818e~~
Current status: https://gist.github.com/EgorBo/827c98cfbbc6c205f5809b3a94c3d480
(24 issues left)

### CODE REVIEW

File: mcs/class/System.Core/common_System.Core.dll.sources [hunk 1]
```diff
@@ -98,6 +98,10 @@ System.Security.Cryptography/SHA512CryptoServiceProvider.cs
 ../../../external/corefx/src/System.Collections/src/System/Collections/Generic/HashSetEqualityComparer.cs
 ../../../external/corefx/src/System.Collections/src/System/Collections/Generic/ICollectionDebugView.cs
 
+
+../../../external/corefx/src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs
```
Comment (EgorBo) [inline mcs/class/System.Core/common_System.Core.dll.sources]: Not sure it should be in System.Core but `DispatchProxyGenerator` uses System.Linq.

**Comment by EgorBo** [comment]
Brakes `mcs` but I know how to fix.

File: mcs/class/System/System.Net.Security/SslStream.cs [hunk 2]
```diff
@@ -721,5 +721,6 @@ public override void EndWrite (IAsyncResult asyncResult)
 			throw new PlatformNotSupportedException (EXCEPTION_MESSAGE);
 		}
 #endif
+		public SslApplicationProtocol NegotiatedApplicationProtocol => throw new PlatformNotSupportedException();
```
Comment (EgorBo) [inline mcs/class/System/System.Net.Security/SslStream.cs]: Martin said we won't support it.

**Comment by marek-safar** [comment]
@EgorBo why was CryptographicOperations type removed?

**Comment by EgorBo** [comment]
@marek-safar 
`CryptographicOperations` and `FileSystemName` existed in both mscorlib and System (were public in both). Do I need to add some sort of TypeForwarder?
The reason why I deleted - I could not add `CryptographicOperations` to netstandard - it could not resolve the assembly

**Comment by marek-safar** [comment]
@monojenkins build failed

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins commit csproj and apidiff

**Comment by EgorBo** [comment]
@marek-safar I suspect `Linux ARMv7 Interpreter` failure is related to PR but I can't figure out why

**Comment by marek-safar** [comment]
oh, I missed that. @lewurm could you help?

**Comment by lewurm** [comment]
will likely be fixed by https://github.com/mono/mono/pull/12985/

**Pull request merged**